iptables/feature_detect.go

```diff
@@ -149,3 +150,74 @@ func (d *FeatureDetector) getKernelVersion() *version.Version {
 	}
 	return kernVersion
 }
+
+// GetIptablesBackend attempts to detect the iptables backend being used where Felix is running.
+// This code is duplicating the detection method found at
+// https://github.com/kubernetes/kubernetes/blob/623b6978866b5d3790d17ff13601ef9e7e4f4bf0/build/debian-iptables/iptables-wrapper#L28
+// If there is a specifiedBackend then it is used but if it does not match the detected
+// backend then a warning is logged.
+func GetIptablesBackend(lookPath func(file string) (string, error), newCmd cmdFactory, specifiedBackend string) string {
```

Comment:
Maybe call this `DetectBackend`?  `Detect` since this function does some work rather than just returning an existing value and removing `Iptables` since it's part of the package name.